### PR TITLE
Enforce workspace unique SourceSetID

### PR DESCRIPTION
**Pull request by sellmair**

### CODE REVIEW

File: runners/cli/build.gradle.kts [hunk 1]
```diff
@@ -10,7 +10,6 @@ repositories {
 
 dependencies {
     implementation("org.jetbrains.kotlinx:kotlinx-cli-jvm:0.2.1")
-    implementation("com.google.code.gson:gson:2.8.5")
```
Comment (BarkingBad) [inline runners/cli/build.gradle.kts:13]: Why do we remove gson dependency for CLI?

Comment (BarkingBad) [reply runners/cli/build.gradle.kts:13]: Oh never mind, I missed you actually made way to extract config from json

File: runners/gradle-plugin/src/main/kotlin/org/jetbrains/dokka/gradle/configurationImplementations.kt [hunk 2]
```diff
@@ -25,64 +33,115 @@ class GradleSourceRootImpl : SourceRoot, Serializable {
...
+    override val sourceSetID: DokkaSourceSetID = DokkaSourceSetID(project, name)
+
     @Input
     override var sourceRoots: MutableList<SourceRoot> = mutableListOf()
+
     @Input
-    override var dependentSourceSets: MutableList<String> = mutableListOf()
+    override var dependentSourceSets: MutableSet<DokkaSourceSetID> = mutableSetOf()
+
+    // TODO NOW: support adding dependencies by name
```
Comment (kamildoleglo) [inline runners/gradle-plugin/src/main/kotlin/org/jetbrains/dokka/gradle/configurationImplementations.kt:60]: Leftover?

Comment (sellmair) [reply runners/gradle-plugin/src/main/kotlin/org/jetbrains/dokka/gradle/configurationImplementations.kt]: Yes! Fixed ✅

**Pull request merged**